components/kyma-environment-broker/internal/runtime/provider_test.go

```diff
@@ -110,55 +155,79 @@ func TestRuntimeComponentProviderGetFailures(t *testing.T) {
 			},
 			returnStatusCode: http.StatusInternalServerError,
 			tempError:        true,
-			expErrMessage:    "while getting open source kyma components: while checking response status code for Kyma components list: got unexpected status code, want 200, got 500, url: https://storage.googleapis.com/kyma-development-artifacts/main-123123/kyma-installer-cluster.yaml, body: Internal Server Error",
+			expErrMessage:    "while getting open source kyma components: while checking response status code for Kyma components list: got unexpected status code, want 200, got 500, url: https://storage.googleapis.com/kyma-development-artifacts/main-123123/kyma-components.yaml, body: Internal Server Error",
 		},
 	}
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
 			// given
-			fakeHTTPClient := newTestClient(t, tc.given.httpErrMessage, tc.returnStatusCode)
+			fakeHTTPClient := newTestClient(t, tc.returnStatusCode, tc.given.httpErrMessage)
 
-			listProvider := runtime.NewComponentsListProvider(tc.given.managedRuntimeComponentsYAMLPath).
+			decider := &ListDecider{newComponentList: true}
+			listProvider := runtime.NewComponentsListProvider(decider, tc.given.managedRuntimeComponentsYAMLPath).
 				WithHTTPClient(fakeHTTPClient)
 
 			// when
 			components, err := listProvider.AllComponents(tc.given.kymaVersion)
 
 			// then
-			assert.Nil(t, components)
+			require.NotNil(t, components)
+			assert.Len(t, components.Components, 0)
 			assert.EqualError(t, err, tc.expErrMessage)
 			assert.Equal(t, tc.tempError, kebError.IsTemporaryError(err))
 		})
 	}
 }
 
+type ListDecider struct {
+	newComponentList bool
+}
+
+func (ld *ListDecider) IsNewComponentList(version string) (bool, error) {
+	return ld.newComponentList, nil
+}
```

Comment:
can we use the ficture you've already provided in `fixture.go`?:
```golang
type FakeListDecider struct {
	New bool
}
```

if there is a reason for redeclaring this fake logic here, you can remove unused parameter 'version' in IsNewComponentList method